libs/langchain/langchain/schema/output_parser.py

```diff
@@ -27,6 +28,20 @@ def parse_result(self, result: List[Generation]) -> T:
             Structured output.
         """
 
+    async def aparse_result(self, result: List[Generation]) -> T:
+        """Parse a list of candidate model Generations into a specific format.
+
+        Args:
+            result: A list of Generations to be parsed. The Generations are assumed
+                to be different candidate outputs for a single model input.
+
+        Returns:
+            Structured output.
+        """
+        return await asyncio.get_running_loop().run_in_executor(
```

Comment:
Shouldn't this also be an abstract method?

Reply:
Ahh okay I think i understand now its running in a thread and resolving the blocking io

Reply:
Yes, and subclasses that actually have a real async implementation can override this method